Dockerfile

```diff
@@ -0,0 +1,126 @@
+####################
+FROM us.icr.io/cdt-common-rns/base-images/ubi8-liberty:latest
+
+# The ARG maintainer is expected to be replaced by a value from the KeyProtect
+# instance configured for the starter app.
+# This integration depends on code in preDockerBuild.sh retrieving the value,
+# setting it as an ENV var and REPLACEARGS: "true" being set in the CI: section
+# of pipeline.config.
+# The keyprotect + replaceargs support is available starting in stable-3.0.2,
+# see https://github.ibm.com/whc-toolchain/whc-commons/tree/stable-3.0.4/docs/ready/ci-vault for more information.
+#ARG WH_COHORTING_APP_TOOLCHAIN_MAINTAINER=cohortTeamFunctionalIdReplacedByValueInCloudKeyProtect
+
+# Labels - certain labels are required if you want to have
+#          a Red Hat certified image (this is not a full set per se)
+LABEL maintainer="IBM Quality Measure and Cohort Service Team" \
+      description="Quality Measure and Cohort Service" \
+      name="cohorting-app" \
+      vendor="Alvearie Open Source by IBM" \
+      version="1.0" \
+      release="1" \
+      summary="Quality Measure and Cohort Service" \
+      description="Quality Measure and Cohort Service available via REST API"
+
+ENV WLP_HOME=/opt/ibm/wlp \
+    SERVER_NAME=myServer \
+    ALVEARIE_HOME=/opt/alvearie \
+    COHORT_DIST_SOLUTION=/app/cohortSolutionDistribution \
+    COHORT_TEST_SOLUTION=/app/cohortTestDistribution \
+    JAVA_HOME=/opt/ibm/java
+ENV ALVEARIE_TEST_HOME=$ALVEARIE_HOME/cohortTestResources \
+    ANT_HOME=$ALVEARIE_HOME/ant
+
+# create server instance
+RUN $WLP_HOME/bin/server create $SERVER_NAME && \
+    mkdir -p $WLP_HOME/usr/servers/$SERVER_NAME/resources/security && \
+    mkdir -p $WLP_HOME/usr/servers/$SERVER_NAME/properties
+
+USER root
+# Update image to pick up latest security updates
+# Make dir for test resources
+# Update symlnk used by Liberty to new server.  Need root.
+RUN microdnf update -y && rm -rf /var/cache/yum && \
+    microdnf install -y --nodocs vim && \
+    microdnf clean all && \
+    mkdir -p $ALVEARIE_TEST_HOME && \
+    ln -sfn $WLP_HOME/usr/servers/$SERVER_NAME /config
+
+#Copy in war files, config files, etc. to final image
+USER whuser
+COPY --from=builder $COHORT_DIST_SOLUTION/solution/webapps/*.war /config/apps/
+COPY --from=builder $COHORT_DIST_SOLUTION/solution/bin/server.xml /config/
+COPY --from=builder $COHORT_DIST_SOLUTION/solution/bin/jvm.options /config/
+COPY --from=builder $COHORT_TEST_SOLUTION/ $ALVEARIE_TEST_HOME/
+COPY --from=builder $ANT_HOME $ANT_HOME
+
+# Add ant to the path so that our test executions
+# can find it
+ENV PATH="$JAVA_HOME/jre/bin:${PATH}:$ANT_HOME/bin"
+
+# Copy our startup script into the installed Liberty bin
+COPY --from=builder $COHORT_DIST_SOLUTION/solution/bin/runServer.sh $WLP_HOME/bin/
+
+# Change to root so we can do chmods to our WH user
+USER root
+
+# Grant write access to apps folder and startup script
+RUN chmod -R u+rwx,g+rwx,o+rwx $WLP_HOME && \
```

Comment:
777 permissions is generally a bad idea. I don't know what the standard is for docker containers and what user we get when we do something like ``kubectl exec -it ... --bash``, but in theory it would be better to grant write+execute to only the whuser account.

Reply:
I had copied this from a Dockerfile written by another team using a liberty server I was using as a template, but your point is well taken. I think granting access just for whuser is better. I will give it a try and see if I can get that to work.